### DISCUSSION TITLE
As sunbird portal is removing the dependency of ngx-infinite-scroll

### DISCUSSION THREAD

**Discussion by rajeevsathish** [samagra-comms/community#26] [General Discussion]
As sunbird portal is removing the dependency of ngx-infinite-scroll and this plugin is used in the @samagra-x/uci-console) 

In sunbird client UI angular application we are removing the plugin ngx-infinate-scroll as this plugin is not updated from long time and is using only angular version.
while removing the plugin from sunbirdEd-portal we found that one of the other plugin is directly using the ngx-infinite-scroll as we are removing the plugin we have removed this plugin also please fix this issue so that sunbirdEd-portal to use the plugin. Remove the ngx-infinte-scroll dependencies from portal code base update the plugin uci-component ("@samagra-x/uci-console": "1.3.7",)

Please plan to remove the plugin ASAP so that we need to take the update to sunbird-portal

cc:  @ChakshuGautam @abhay-samagra  @Krishnaj20 @vinukumar-vs

**Comment by chitranshu-keshav** [comment]
@ChakshuGautam @abhay-samagra @Naveenkumar739 As informed above by @rajeevsathish the plugin "ngx-infinite-scroll" is being removed from Sunbird ED on account of Angular migration to version 14. This plugin is using a very old angular version hence Sunbird ED is removing it.

This plugin is used by UCI also and an alternate for the same has to be implemented otherwise ED will not be able to use UCI. Similarly since ED is moving to angular version 14, unless UCI is upgraded to angular 14, Sunbird ED will not be able to use UCI. 

If the above is not taken care, the UCI functionality will not be available in the Sunbird ED in release 6.0.0 and beyond. This might have a major impact on the Sunbird ED adopters who are using this functionality. The release development is planned to be complete by 5th May 2023 and it would be great if these changes are done by that time. Release timelines of Sunbird ED can be accessed [here](https://ed.sunbird.org/use/releases/release-calendars/release-calendar-6.0.0) 

Can you please get back on the plan for above changes and by when it can be done. Will appreciate a response on this thread

cc: @Krishnaj20 @vinukumar-vs @madhucr @kameshbhr

**Reply by vinukumar-vs** [reply]
@ChakshuGautam @abhay-samagra 
Any update on this. Without UCI upgrading to Angular 14, Sunbird-ED can't go for prod(release-6.0.0). Which is a blocker to release Sunbird-ED 6.0

Please call our if you need any help.
@Krishnaj20 @rajeevsathish @madhucr @alok-os @vrayulu

**Comment by abhay-samagra** [comment]
Hi Vinu,Let me get back after checking internally on this.RegardsAbhayOn 20-Apr-2023, at 11:54 AM, vinukumar-vs ***@***.***> wrote:﻿
@ChakshuGautam @abhay-samagra
Any update on this. Without UCI upgrading to Angular 14, Sunbird-ED can't go for prod(release-6.0.0). Which is a blocker to release Sunbird-ED 6.0
Please call our if you need any help.
@Krishnaj20 @rajeevsathish @madhucr @alok-os @vrayulu

**Comment by rajeevsathish** [comment]
@abhay-samagra can you please create branches for angular migration as our team has forked your repo and have migrated till Angular 13 we will raise PR and contribute for the same
cc: @vinukumar-vs @chitranshu-keshav @Krishnaj20

**Comment by abhay-samagra** [comment]
Hi SathishManoj from our side is going to work on this ***@***.*** please take a note of the trail mail.RegardsAbhayOn 21-Apr-2023, at 11:50 AM, Rajeev Sathish ***@***.***> wrote:﻿
@abhay-samagra can you please create branches for angular migration as our team has forked your repo and have migrated till Angular 13 we will raise PR and contribute for the same

**Comment by abhishekpnt** [comment]
@abhay-samagra Can you please create a branch "release-6.0.0_v13" in the repo https://github.com/samagra-comms/uci-console so that we can push our changes to this branch and raise the PR
CC: @vinukumar-vs @rajeevsathish @rajeshkumaravel

**Comment by abhay-samagra** [comment]
@abhishekpnt - here it is.
https://github.com/samagra-comms/uci-console/tree/release-6.0.0_v13

**Reply by abhishekpnt** [reply]
@abhay-samagra 

I have raised a PR for Angular-13 changes. Please review and merge it to the same branch and publish the package. https://github.com/samagra-comms/uci-console/pull/37

Also can you please create one more branch "release-6.0.0_v14" from "release-6.0.0_v13" after merging  in the same repo for Angular-14 changes?

CC: @vinukumar-vs @rajeevsathish @rajeshkumaravel

**Reply by abhishekpnt** [reply]
@abhay-samagra 

Any update on this?

@vinukumar-vs @rajeevsathish @rajeshkumaravel @chitranshu-keshav

**Reply by abhay-samagra** [reply]
@abhishekpnt - Sorry, missed out your message. 
Here is the branch for Angular 14 changes: 
https://github.com/samagra-comms/uci-console/tree/release-6.0.0_v14
I have only superficially reviewed the code, please ensure it works before committing.

**Reply by abhishekpnt** [reply]
@abhay-samagra 

Can you please review and approve this PR and publish the package
https://github.com/samagra-comms/uci-console/pull/38

CC: @vinukumar-vs @rajeevsathish @rajeshkumaravel @chitranshu-keshav

**Reply by abhishekpnt** [reply]
@abhay-samagra Please publish the package and let me know once it is done

**Reply by abhay-samagra** [reply]
done.

**Reply by abhishekpnt** [reply]
@abhay-samagra  can you please provide me link to the npm package

**Reply by abhay-samagra** [reply]
https://github.com/samagra-comms/uci-console/releases/tag/6.0.0_v14

**Reply by abhishekpnt** [reply]
@abhay-samagra 

I dont see the latest version here: https://www.npmjs.com/package/@samagra-x/uci-console

Can you please publish the package to npm so that we can use it in our codebase

CC: @vinukumar-vs @rajeevsathish @rajeshkumaravel @chitranshu-keshav

**Reply by abhishekpnt** [reply]
@abhay-samagra 

Any update on npm package?

CC: @vinukumar-vs @rajeevsathish @rajeshkumaravel

**Comment by abhishekpnt** [comment]
@abhay-samagra

Any update on npm package?

CC: @vinukumar-vs @rajeevsathish @rajeshkumaravel @chitranshu-keshav

**Comment by abhay-samagra** [comment]
Hey Abhishek,I am on an internal training since Sunday till today. Will share the package here tomorrow.Sent from my iPhoneOn 02-May-2023, at 11:56 AM, Abhishek P N ***@***.***> wrote:﻿
@abhay-samagra
Any update on npm package?
CC: @vinukumar-vs @rajeevsathish @rajeshkumaravel @chitranshu-keshav

**Comment by abhay-samagra** [comment]
Hi @abhishekpnt 
Here's the package link: https://www.npmjs.com/package/@samagra-x/uci-console
